purldb-toolkit/src/purldb_toolkit/purlcli.py

```diff
@@ -27,47 +29,129 @@ def purlcli():
     type=click.File(mode="w", encoding="utf-8"),
     required=True,
     default="-",
-    help="Write validation output as JSON to FILE.",
+    help="Write versions output as JSON to FILE.",
 )
 @click.option(
     "--file",
     type=click.File(mode="r", encoding="utf-8"),
     required=False,
     help="Read a list of PURLs from a FILE, one per line.",
 )
-def validate(purls, output, file):
+def get_meta(purls, output, file):
     """
-    Check the syntax of one or more PURLs.
+    Given one or more PURLs, for each PURL, return a mapping of metadata
+    fetched from the fetchcode package.py info() function.
     """
-    if purls and file:
-        raise click.UsageError("Use either purls or file but not both.")
-
-    if not (purls or file):
-        raise click.UsageError("Use either purls or file.")
+    check_for_duplicate_input_sources(purls, file)
 
     if file:
         purls = file.read().splitlines(False)
 
-    validated_purls = validate_purls(purls)
-    json.dump(validated_purls, output, indent=4)
+    context = click.get_current_context()
+    manual_command_name = context.command.name
 
+    meta_info = get_meta_details(purls, output, file, manual_command_name)
+    json.dump(meta_info, output, indent=4)
+
+
+def get_meta_details(purls, output, file, manual_command_name):
+    """
+    Return a dictionary containing metadata for each PURL in the `purls` input
+    list.  `check_meta_purl()` will print an error message to the console
+    (also displayed in the JSON output) when necessary.
+    """
+    context_purls = [p for p in purls]
+    context_file = file
+    context_file_value = None
+    if context_file:
+        context_file_value = context_file.name
+
+    meta_details = {}
+    meta_details["headers"] = []
+    meta_details["packages"] = []
+
+    headers_content = {}
```

Comment:
I would put all the code related to header creation into another function so that you can use it in other commands. This will help ensure that we create headers in a consistent fashion.

Reply:
Done -- thanks @JonoYang .